### DISCUSSION TITLE
scrollbars not visible

### DISCUSSION THREAD

**Discussion by kayetter** [idiotWu/smooth-scrollbar#351] [Q&A] [answered]
scrolling works with mousewheel but the scrollbar is not visible. I am using the attribute method. I tried testing with putting attribute 'scrollbar' on body. And while the scrolling works, i do not see the scrollbar. I tried adjusting the z-index in css but this did not work either

## Expected Behavior
the vertical scroll bar appears as soon as i start scrolling

## Current Behavior
scrollbar never appears

## Environment
i am using chrome on a surface but it does not work in firefox either. i am using php and wamp server

My div has a fixed position on the page and my jquery is as follows
```
 if (width > 740) {
    $(".body-content").attr("scrollbar","");
  }
```

    <section class="body-content">

        <?php include("portal.php") ?>

    </section>
`
i fell like i am missing something simple

**Comment by dolphin-wood** [comment]
Try

```js
if (width > 740) {
    Scrollbar.init($(".body-content").get(0))
}
```

**Comment by kayetter** [comment]
thanks for your prompt reply. I tried as you suggested and it still failed to show the scrollbar. Unfortunately, your plugin and the jquery.scrollTo plugin are interacting. I uninstalled the jquery.scrollTo plugin and again, the suggestion you offered did not work. i tried with and without the .get. It is all the same behavior where the scrolling works but the scrollbar is not visible. 

I like your plugin but i may have to move on to the next custom scrollbar option.

**Comment by dolphin-wood** [comment]
Can you provide a link to the repo or an online demo so that I could find out what's happening?

**Comment by kayetter** [comment]
Yes. here is the repository. [https://github.com/kayetter/bodymindbusiness.com.git](url)
How do i use the scrollbar.getSize(); i keep getting the following error so i think my syntax is wrong: "uncaught TypeError: Cannot read property 'getSize' of undefined" How do i define it?

I tried the following: 
`  var scrollbar = Scrollbar.init($(".body-content").get(0));
    scrollbar.scrollbar.getSize();`

**Comment by kayetter** [comment]
nevermind, i figured the getSize out using the following:
`  var scrollbar = Scrollbar.init($(".body-content").get(0));
    console.log(scrollbar.getSize(scrollbar));`

**Comment by dolphin-wood** [comment]
Thanks for the link. Will have a look after school.

The term `scrollbar` in docs means the instance you get from `Scrollbar.init()`. So you just need to call `scrollbar.getSize()`.

**Comment by kayetter** [comment, marked as answer]
ok, you are probably going to kill me but i figured out that it was all because the .css was not importing into my library file during my bower task in gulp. Thanks for looking into it. 

I was wondering though, is your smoothscroll imcompatible with jquery-scrollTo? would these two plugins interfere? If so, does your scrollbar.setPostion or scrollbar.Target allow me to target scrolling to a specific element. 

I use scrollTo as a click event to get me to the right anchor ID. when i am using in conjunction with smooth scrollbar the div migrates to the anchor and then instantly pops back up to the top of the container div which is not desired behavior. I would like the div to stay at the specified anchor target.

**Comment by dolphin-wood** [comment]
Actually you can use the built-in `scrollbar.scrollTo()`or `.scrollIntoView()` method.
